contract/models/res_partner.py

```diff
@@ -7,30 +7,63 @@
 class ResPartner(models.Model):
     _inherit = 'res.partner'
 
-    contract_count = fields.Integer(
-        string='Contracts',
+    sale_contract_count = fields.Integer(
+        string='Sale Contracts',
+        compute='_compute_contract_count',
+    )
+    purchase_contract_count = fields.Integer(
+        string='Purchase Contracts',
         compute='_compute_contract_count',
     )
 
     def _compute_contract_count(self):
-        Contract = self.env['account.analytic.account']
+        contract_model = self.env['account.analytic.account']
         today = fields.Date.today()
         for partner in self:
-            partner.contract_count = Contract.search_count([
+            partner.sale_contract_count = contract_model.search_count([
```

Comment:
This is not very performance wise. Please implement it through a `read_group` outside the loop with lazy=False, the domain without the contract_type, and grouping by partner_id and contract_type, and you'll get the same results but faster.

Reply:
and what about <code>('partner_id', 'child_of', partner.ids),</code>?

Reply:
Uhm, interesting. You can add `commercial_partner_id` field and group by it (or even better, search directly by it).

Reply:
But it's not the same. That will give different result

Reply:
Why? You shouldn't have "grandsons" (Odoo doesn't allow it naturally). Any way, the initial intention of that was to cover all the hierarchy.

Reply:
That's not what I mean. I mean that now the search is counting the partner contracts and their child contracts. But not their parent contracts. Thus, if I use the commercial_partner_id, for child partners, they will have in the count their parent also.